main/src/ca/uwaterloo/flix/language/errors/StratificationError.scala

```diff
@@ -17,14 +17,33 @@
 package ca.uwaterloo.flix.language.errors
 
 import ca.uwaterloo.flix.language.CompilationError
+import ca.uwaterloo.flix.language.ast.TypedAst.Predicate.Head
+import ca.uwaterloo.flix.language.ast.TypedAst.Predicate.Head.{False, True}
 import ca.uwaterloo.flix.language.ast._
+import ca.uwaterloo.flix.util.vt.VirtualString._
 import ca.uwaterloo.flix.util.vt.VirtualTerminal
 
 /**
   * An error raised to indicate that a property is violated.
   */
-case class StratificationError(constraints: List[TypedAst.Constraint]) extends CompilationError {
+case class StratificationError(constrs: List[TypedAst.Constraint]) extends CompilationError {
   val kind: String = "Stratification Error"
-  val source: SourceInput = constraints.head.loc.source
-  val message: VirtualTerminal = ??? // TODO
+  val source: SourceInput = constrs.head.loc.source
+  val message: VirtualTerminal = {
+    val vt = new VirtualTerminal
+    vt << Line(kind, source.format) << NewLine
+    vt << ">> Stratification Error in Constraints" << NewLine
+
+    vt << (if (constrs.size > 1) {"Consider the rules:"} else {"Consider the rule:"}) << NewLine
+    constrs.foreach(rule => {
+      vt << (rule.head match {
+        case True(loc) => loc.lineAt(loc.beginLine)
```

Comment:
I've used `loc.format` instead now. There error message now looks like

```
-- Stratification Error -------------------------------------------------- foo.flix

>> Stratification Error in Constraints
The following constraints form a negative cycle at locations
	foo.flix:6:1
	foo.flix:13:1
```
I'd still prefer it to print the actual code, since that would make it really easy to see the cycle, but it does seem confusing to only print the first line if it spans multiple lines. Do we not currently have a way to pretty print ast nodes?